From ab63493596dcdd7057be2be4d5241d2a9fee37d8 Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Mon, 16 Jan 2017 11:00:11 +0100
Subject: [PATCH] return proper version information

Some tools and libraries like libepoxy rely on useful version
information.

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 src/glx/glx.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/glx/glx.c b/src/glx/glx.c
index d436bae..00dd1fd 100644
--- a/src/glx/glx.c
+++ b/src/glx/glx.c
@@ -532,7 +532,12 @@ const char *glXQueryExtensionsString(Display *dpy, int screen) {
 
 const char *glXQueryServerString(Display *dpy, int screen, int name) {
     PROXY_GLES(glXQueryServerString);
-    return "";
+    switch (name) {
+        case GLX_VENDOR: return "Ryan Hileman (lunixbochs)";
+        case GLX_VERSION: return "1.4 glshim";
+        case GLX_EXTENSIONS: return glXQueryExtensionsString(dpy, 0);
+    }
+    return NULL;
 }
 
 Bool glXQueryExtension(Display *display, int *errorBase, int *eventBase) {
@@ -556,13 +561,12 @@ Bool glXQueryVersion(Display *dpy, int *maj, int *min) {
 
 const char *glXGetClientString(Display *display, int name) {
     PROXY_GLES(glXGetClientString);
-    // TODO: return actual data here
     switch (name) {
-        case GLX_VENDOR: break;
-        case GLX_VERSION: break;
-        case GLX_EXTENSIONS: break;
+        case GLX_VENDOR: return "Ryan Hileman (lunixbochs)";
+        case GLX_VERSION: return "1.4 glshim";
+        case GLX_EXTENSIONS: return glXQueryExtensionsString(display, 0);
     }
-    return "";
+    return NULL;
 }
 
 // stubs for glfw (GLX 1.3)
-- 
2.7.4

